aws/table_aws_waf_rate_based_rule.go

```diff
@@ -0,0 +1,223 @@
+				Transform:   transform.FromField("MatchPredicates"),
+			},
+			{
+				Name:        "rate_key",
+				Description: "The name or description for the metrics for a RateBasedRule.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsWafRateBasedRules,
+			},
+			{
+				Name:        "rate_limit",
+				Description: "The name or description for the metrics for a RateBasedRule.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsWafRateBasedRules,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the Rule.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listAwsWafRateBasedRuleTags,
+				Transform:   transform.FromField("TagInfoForResource.TagList"),
+			},
+
+			// // Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listAwsWafRateBasedRuleTags,
+				Transform:   transform.FromField("TagInfoForResource.TagList").Transform(tagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsWafRateBasedRuleAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+func listAwsWafRateBasedRules(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listAwsWafRateBasedRules")
+	// Create session
+	svc, err := WafService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	// List call
+	params := &waf.ListRateBasedRulesInput{}
+	pagesLeft := true
+	for pagesLeft {
+		response, err := svc.ListRateBasedRules(params)
+		if err != nil {
+			return nil, err
+		}
+		for _, rule := range response.Rules {
+			d.StreamListItem(ctx, rule)
+		}
+		if response.NextMarker != nil {
+			pagesLeft = true
+			params.NextMarker = response.NextMarker
+		} else {
+			pagesLeft = false
+		}
+	}
+	return nil, nil
+}
+
+////  HYDRATE FUNCTIONS
+func getAwsWafRateBasedRules(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsWafRateBasedRules")
+	var id string
+	if h.Item != nil {
+		id = rateBasedRuleData(h.Item)
+	} else {
+		id = d.KeyColumnQuals["rule_id"].GetStringValue()
+	}
+	// Create Session
+	svc, err := WafService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	// Build the params
+	params := &waf.GetRateBasedRuleInput{
+		RuleId: &id,
+	}
+	// Get call
+	data, err := svc.GetRateBasedRule(params)
+	if err != nil {
+		logger.Debug("getAwsWafRateBasedRules", "ERROR", err)
+		return nil, err
+	}
+	return data.Rule, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
```

Comment:
are listAwsWafRateBasedRuleTags and getAwsWafRateBasedRuleAkas transform functions?

Reply:
Yes

Reply:
No , these are not transform function.